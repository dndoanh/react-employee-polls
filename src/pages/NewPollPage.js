import { useState } from "react";
import { connect } from "react-redux";
import { useNavigate } from "react-router-dom";
import { handleAddQuestion } from "../actions/questions";
import { Button, Card, Container, Form } from "react-bootstrap";

const NewPollPage = ({ dispatch }) => {
  const navigate = useNavigate();
  const [options, setOptions] = useState({
    firstOption: "",
    secondOption: "",
  });
  const handleSubmit = (e) => {
    e.preventDefault();
    if (options.firstOption === "" || options.secondOption === "") {
      alert("Please input for both options");
      return;
    }
    dispatch(handleAddQuestion(options.firstOption, options.secondOption));
    navigate("/");
  };
  return (
    <>
      <Container>
      <h1 className="text-center">New Poll</h1>
        <Card className="text-center">
          <Card.Header>Would You Rather</Card.Header>
          <Card.Title>Create Your Own Poll</Card.Title>
          <Card.Body>
            <Form onSubmit={handleSubmit}>
              <Form.Group className="mb-3">
                <Form.Label>First Option</Form.Label>
                <Form.Control
                  type="text"
                  placeholder="Option One"
                  value={options.firstOption}
                  onChange={(e) =>
                    setOptions({ ...options, firstOption: e.target.value })
                  }
                />
              </Form.Group>
              <Form.Group className="mb-3">
                <Form.Label>Second Option</Form.Label>
                <Form.Control
                  type="text"
                  placeholder="Option Two"
                  value={options.secondOption}
                  onChange={(e) =>
                    setOptions({ ...options, secondOption: e.target.value })
                  }
                />
              </Form.Group>
              <Button variant="primary" type="submit">
                Submit
              </Button>
            </Form>
          </Card.Body>
        </Card>
      </Container>
    </>
  );
};

export default connect()(NewPollPage);
