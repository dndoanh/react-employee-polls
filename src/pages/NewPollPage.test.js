import { fireEvent, render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import React from "react";
import NewPollPage from "./NewPollPage";
import { createStore } from "redux";
import reducer from "../reducers";
import middleware from "../middlewares";

const store = createStore(reducer, middleware);

describe("NewPollPage", () => {
  test("should render all components", () => {
    const view = render(
      <Provider store={store}>
        <BrowserRouter>
          <NewPollPage />
        </BrowserRouter>
      </Provider>
    );
    expect(view).toBeDefined();
    expect(view).toMatchSnapshot();
  });

  test("should display all elements", () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <NewPollPage />
        </BrowserRouter>
      </Provider>
    );
    const options = { optionOne: "New Jersey", optionTwo: "Alabama" };
    const firstOptionInputElement = screen.getByTestId("option-one-input");
    const secondOptionInputElement = screen.getByTestId("option-two-input");
    const submitButtonElement = screen.getByTestId("btn-submit-poll");

    fireEvent.change(firstOptionInputElement, {
      target: { value: options.optionOne },
    });
    fireEvent.change(secondOptionInputElement, {
      target: { value: options.optionTwo },
    });
    
    expect(submitButtonElement.textContent).toBe("Submit");
    expect(firstOptionInputElement.value).toBe(options.optionOne);
    expect(secondOptionInputElement.value).toBe(options.optionTwo);
  });
});
