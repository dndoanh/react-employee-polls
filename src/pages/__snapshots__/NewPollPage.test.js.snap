// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPollPage should render all components 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <h1
          class="text-center"
        >
          New Poll
        </h1>
        <div
          class="text-center card"
        >
          <div
            class="card-header"
          >
            Would You Rather
          </div>
          <div
            class="card-title h5"
          >
            Create Your Own Poll
          </div>
          <div
            class="card-body"
          >
            <form
              class=""
            >
              <div
                class="mb-3"
              >
                <label
                  class="form-label"
                >
                  First Option
                </label>
                <input
                  class="form-control"
                  data-testid="option-one-input"
                  placeholder="Option One"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="mb-3"
              >
                <label
                  class="form-label"
                >
                  Second Option
                </label>
                <input
                  class="form-control"
                  data-testid="option-two-input"
                  placeholder="Option Two"
                  type="text"
                  value=""
                />
              </div>
              <button
                class="btn btn-primary"
                data-testid="btn-submit-poll"
                type="submit"
              >
                Submit
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <h1
        class="text-center"
      >
        New Poll
      </h1>
      <div
        class="text-center card"
      >
        <div
          class="card-header"
        >
          Would You Rather
        </div>
        <div
          class="card-title h5"
        >
          Create Your Own Poll
        </div>
        <div
          class="card-body"
        >
          <form
            class=""
          >
            <div
              class="mb-3"
            >
              <label
                class="form-label"
              >
                First Option
              </label>
              <input
                class="form-control"
                data-testid="option-one-input"
                placeholder="Option One"
                type="text"
                value=""
              />
            </div>
            <div
              class="mb-3"
            >
              <label
                class="form-label"
              >
                Second Option
              </label>
              <input
                class="form-control"
                data-testid="option-two-input"
                placeholder="Option Two"
                type="text"
                value=""
              />
            </div>
            <button
              class="btn btn-primary"
              data-testid="btn-submit-poll"
              type="submit"
            >
              Submit
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
